### ISSUE TITLE
Ensure that `ExperimentSeries` don't point to deleted experiments

### ISSUE THREAD

**Issue by BeritJanssen** [Amsterdam-Music-Lab/MUSCLE#866]
This is again a bit of a bug waiting to happen. For practical reasons (a model cannot have several `ManyToMany` fields pointing to the same model), the `first_experiments`, `random_experiments` and `last_experiments` fields are now saved as lists of `pk`s of experiments which *at the point of creation of the ExperimentSeries* exist in the database. However, if any of the associated experiments were to be deleted, the `ExperimentSeries` would run into a bug mid-way.

**To (re)produce**
Create an ExperimentSeries, associate experiments with it, then delete the experiment. Run the ExperimentSeries.

**Possible solution**
We might introduce an extra hook when deleting an `Experiment` object, which will delete their pks from the `ExperimentSeries` objects.

**Alternative solution**
We could also catch this bug at run-time: if a pk in a requested `ExperimentSeries` doesn't exist, just skip to the next experiment.

**Comment by BeritJanssen** [comment]
Nice idea! Perhaps we could even have the lists of experiments themselves in an extra model, i.e.,
```python
class ExperimentGroup(models.Model):
   series = models.ForeignKey(ExperimentSeries, on_delete=models.CASCADE)
   order = models.PositiveIntegerField(default=0)
   dashboard = models.BooleanField()
   randomize = models.BooleanField()
  
class ExperimentSeriesExperiment(mdoels.Model):
   (...)
   stage = models.ForeignKey(ExperimentGroup, on_delete=models.CASCADE)
 ```
That way, we could give users all flexibility to group different stages together.
That way, we can

**Comment by BeritJanssen** [comment]
Closed in favour of #875 .
